### ISSUE TITLE
Fees

### ISSUE THREAD

**Issue by neroNL29** [annularis/shop#41]
Hey, i installed everything perfect and it seems to work very good for now, when i test vendor fees it generate me payment address, but this payment address is not mine, i using bitcoin-qt 15.0.1 i put my private key in bip32, get my public key and put it the website, but all addresses that website generate are not mine, what can be the reason?

**Comment by danfossi** [comment]
Hi @neroNL29 , I think it is related to our issue with the bitcoin library.

* https://github.com/annularis/shop/issues/14
* https://github.com/annularis/shop/issues/17

**Comment by neroNL29** [comment]
the problem is that i don't get any errors, i receive payment address, i can pay to it... but this address is not mine.

**Comment by serhack** [comment]
It seems a cool bug. Can you see if the unknown wallets are on your db?

**Comment by neroNL29** [comment]
no, i track the code, and i see that this address is generated by my xpub, its really strange, i try everything, but no matter what key i add xpriv or xpub it still continue generating addresses for payments

**Comment by neroNL29** [comment]
| Edit | Copy | Delete | 1 | fees | 1NzmMajue8E3dQxbEkXtzZWft7sQSfMeCs
-- | -- | -- | -- | -- | -- | --
  | Edit | Copy | Delete | 2 | fees | 1MxroAxMydLJ9UFdEh7M1EoQi1CssamuAh
  | Edit | Copy | Delete | 3 | fees | 1BXwP3mD7QtBBmx2nHFZoVd68jNJ4C5fNR
  | Edit | Copy | Delete | 4 | fees | 1LGMQZahtSKxLshorGG9ECiwg3qzFEP8RN
  | Edit | Copy | Delete | 5 | fees | 1D6GB8ZMSbEKCboqZW2oW1qLS1XXxZy17H
  | Edit | Copy | Delete | 6 | fees | 1226GZ47PyZN6HxmMak4qTCWBhHhAcfP2D
  | Edit | Copy | Delete | 7 | fees | 198jiNP5qC7XTD8CqUYDFxYmQrjZdvmL1s
  | Edit | Copy | Delete | 8 | fees | 1Jz6rADXQwmVb26voEJ2Xz6M8YSqsWWzE2
  | Edit | Copy | Delete | 9 | fees | 1C4nRnFhjvXyxLD8VtcAAmDtSsymHL3Atk
  | Edit | Copy | Delete | 10 | fees | 1MxroAxMydLJ9UFdEh7M1EoQi1CssamuAh
  | Edit | Copy | Delete | 11 | fees | 14wKDEEY9xuRAAetWsjEXRrGp53qNbBmJY
  | Edit | Copy | Delete | 12 | fees | 14Uoq3hKavqhW3r4eQ3zk1EUKQQcMjX6Dd

no one of this addresses are mine, the xpub is correct...

**Comment by serhack** [comment]
Great, cc @danfossi what is happening?

**Comment by danfossi** [comment]
Really, I have no idea @serhack , we need to check the bitcoin-lib and eventually update it

**Comment by neroNL29** [comment]
Maybe is something that i don't understand here, but i try with other up to date code not connected with this project, only to generate addresses from xpub, it makes the same addresses as this project, they again are not mine... maybe i do something wrong.
